### PR TITLE
Lt 1116 denial reasons

### PR DESCRIPTION
**Pull request by jfaracik**

### CODE REVIEW

File: drafts/draft_end_user_views.py [hunk 1]
```diff
@@ -7,71 +7,38 @@
...
-
-        end_user_ids = EndUserOnDraft.objects.filter(draft=draft).values_list('end_user', flat=True)
-        end_users = EndUser.objects.filter(id__in=end_user_ids)
-        serializer = EndUserViewSerializer(end_users, many=True)
-        return JsonResponse(data={'end_users': serializer.data})
-
     @transaction.atomic
     def post(self, request, pk):
         organisation = get_organisation_by_user(request.user)
         data = JSONParser().parse(request)
```
[the remark this answers is not in the document]
Comment (jfaracik) [reply drafts/draft_end_user_views.py]: This was intentional in order to add the organisation field to the data. The serialiser still handles all the errors for us.

File: applications/serializers.py [hunk 2]
```diff
@@ -62,22 +69,69 @@ class Meta:
...
     activity = serializers.CharField()
-    status = serializers.CharField() # Doesnt validate yet
-    licence_type = serializers.ChoiceField([(tag.name, tag.value) for tag in LicenceType],
-                                           error_messages={
-                                               'required': 'Select which type of licence you want to apply for.'})
-    export_type = serializers.ChoiceField([(tag.name, tag.value) for tag in ExportType],
-                                          error_messages={
-                                              'required': 'Select if you want to apply for a temporary or permanent '
-                                                          'licence.'})
+    status = serializers.ChoiceField(choices=ApplicationStatus.choices)
```
Comment (5pence) [inline applications/serializers.py]: Why repeat the 3 fields status, license_type and export_type? These are already included in ApplicationBaseSerializer and are therefore superfluous

**Comment by lukesa** [comment]
clean up commits and squash yours

File: applications/models.py [hunk 3]
```diff
@@ -66,14 +48,15 @@ class Destination(models.Model):
 
 
 @reversion.register()
-class EndUserOnApplication(models.Model):
```
Comment (anirbanlahiri-dit) [inline applications/models.py]: Is it necessary to change this as part of this story? Or is it part of a refactoring effort?

File: static/denial_reasons/models.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import reversion
+from django.db import models
+
+
+@reversion.register()
+class DenialReason(models.Model):
+    id = models.CharField(primary_key=True, editable=False, max_length=3)
+    deprecated = models.BooleanField(default=False, null=False, blank=False)
```
Comment (anirbanlahiri-dit) [inline static/denial_reasons/models.py]: Why is it called deprecated?

File: applications/models.py [hunk 5]
```diff
@@ -66,14 +48,15 @@ class Destination(models.Model):
...
 
 @reversion.register()
-class SiteOnApplication(models.Model):
+class ApplicationDenialReason(models.Model):
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    site = models.ForeignKey(Site, related_name='sites_on_application', on_delete=models.CASCADE)
-    application = models.ForeignKey(Application, related_name='application_sites', on_delete=models.CASCADE)
+    application = models.ForeignKey(Application, related_name='application_denial_reason', on_delete=models.CASCADE)
+    reasons = models.ManyToManyField(DenialReason)
+    reasoning = models.TextField(default=None, blank=True, null=True, max_length=2200)
```
Comment (anirbanlahiri-dit) [inline applications/models.py]: Would it be better to name this as reason_details

File: static/denial_reasons/models.py (hunk 4 above)
Comment (anirbanlahiri-dit) [reply static/denial_reasons/models.py]: A more readable field name would be preferred E.g. status_deprecated

Comment (lukesa) [reply static/denial_reasons/models.py]: This is contextual, ask Chris for more details.

**Comment by anirbanlahiri-dit** [comment]
Merged to master

**Pull request closed without merging**